Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -7559,28 +7559,26 @@ class BBQJIT {
 
         constexpr unsigned minCasesForTable = 7;
         if (minCasesForTable <= targets.size()) {
-            Vector<Box<CCallHelpers::Label>> labels;
-            labels.reserveInitialCapacity(targets.size());
             auto* jumpTable = m_callee.addJumpTable(targets.size());
             auto fallThrough = m_jit.branch32(RelationalCondition::AboveOrEqual, wasmScratchGPR, TrustedImm32(targets.size()));
             m_jit.zeroExtend32ToWord(wasmScratchGPR, wasmScratchGPR);
             m_jit.lshiftPtr(TrustedImm32(3), wasmScratchGPR);
             m_jit.addPtr(TrustedImmPtr(jumpTable->data()), wasmScratchGPR);
             m_jit.farJump(Address(wasmScratchGPR), JSSwitchPtrTag);
 
-            for (unsigned index = 0; index < targets.size(); ++index) {
-                Box<CCallHelpers::Label> label = Box<CCallHelpers::Label>::create(m_jit.label());
-                labels.unsafeAppendWithoutCapacityCheck(label);
-                bool isCodeEmitted = currentControlData().addExit(*this, targets[index]->targetLocations(), results);
+            auto labels = WTF::map(targets, [&](auto& target) {
+                auto label = Box<CCallHelpers::Label>::create(m_jit.label());
+                bool isCodeEmitted = currentControlData().addExit(*this, target->targetLocations(), results);
                 if (isCodeEmitted)
-                    targets[index]->addBranch(m_jit.jump());
+                    target->addBranch(m_jit.jump());
                 else {
                     // It is common that we do not need to emit anything before jumping to the target block.
                     // In that case, we put Box<Label> which will be filled later when the end of the block is linked.
                     // We put direct jump to that block in the link task.
-                    targets[index]->addLabel(WTFMove(label));
+                    target->addLabel(Box<CCallHelpers::Label> { label });
```

Comment:
I would have expected `Box { label }` here. Do we really need the template argument?

Reply:
I'll check, thanks.

Reply:
Could also experiment with leaving out the type altogether and just putting in the braces. No idea if that will work.